service-api/module/Application/src/Yoti/YotiService.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Yoti;
+
+use Application\Exceptions\YotiException;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use Laminas\Http\Response;
+
+/**
+ * @psalm-suppress PossiblyUnusedProperty
+ * Suppress unused $client pending implementation
+ */
+class YotiService implements YotiServiceInterface
+{
+    public function __construct(
+        public readonly Client $client
+    ) {
+    }
+
+    /**
+     * @param string $postCode
+     * @return array
+     * Get post offices near the location
+     * @throws GuzzleException
+     * @throws YotiException
+     */
+    public function postOfficeBranch(string $postCode): array
+    {
+        try {
+            $results = $this->client->post('/idverify/v1/lookup/uk-post-office', [
+                'json' => ['SearchString' => $postCode],
+            ]);
+            if ($results->getStatusCode() !== Response::STATUS_CODE_200) {
+                throw new YotiException($results->getReasonPhrase());
+            }
+            return json_decode(strval($results->getBody()), true);
+        } catch (GuzzleException $e) {
+            return ["error" => $e->getMessage()];
```

Comment:
I don't like returning errors in the same shape as successes because it makes it hard to distinguish the two. Maybe it would make more sense to throw e.g. `new YotiException("A connection error occurred", previous: $e)`?